research/lamp-automated-model-parallelism/unet_pipe.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from typing import List
+
+import torch
+from monai.networks.blocks import Convolution, UpSample
+from monai.networks.layers.factories import Act, Conv, Norm
+from torch import nn
+from torchgpipe.skip import Namespace, pop, skippable, stash
+
+
+@skippable(stash=["skip"], pop=[])
+class Stash(nn.Module):
+    def forward(self, input: torch.Tensor):
+        yield stash("skip", input)
+        return input  # noqa  using yield together with return
+
+
+@skippable(stash=[], pop=["skip"])
+class PopCat(nn.Module):
+    def forward(self, input: torch.Tensor):
+        skip = yield pop("skip")
+        if skip is not None:
+            input = torch.cat([input, skip], dim=1)
+        return input
+
+
+def flatten_sequential(module: nn.Sequential):
+    """
+    Recursively make all the submodules sequential.
+
+    Args:
+        module: a torch sequential model.
+    """
+    if not isinstance(module, nn.Sequential):
+        raise TypeError("module must be a nn.Sequential instance.")
+
+    def _flatten(module):
+        for name, child in module.named_children():
+            if isinstance(child, nn.Sequential):
+                for sub_name, sub_child in _flatten(child):
+                    yield f"{name}_{sub_name}", sub_child
+            else:
+                yield name, child
+
+    return nn.Sequential(OrderedDict(_flatten(module)))
+
+
+class DoubleConv(nn.Module):
```

Comment:
I feel `DoubleConv` is a common module in many networks, could you please move it to network/blocks?

Reply:
yes, it is common, this block is only suitable for the two cases used in this research. we'll need another PR to introduce more args properly and make it generic